components/txn_types/src/lock.rs

```diff
@@ -20,13 +20,15 @@ pub enum LockType {
     Put,
     Delete,
     Lock,
-    Pessimistic,
+    // If the pessimistic lock is locked with conflict, the bool values is true.
+    Pessimistic(bool),
 }
 
 const FLAG_PUT: u8 = b'P';
 const FLAG_DELETE: u8 = b'D';
 const FLAG_LOCK: u8 = b'L';
 const FLAG_PESSIMISTIC: u8 = b'S';
+const FLAG_PESSIMISTIC_WTIH_CONFLICT: u8 = b'F';
```

Comment:
> I think more simple approach to do that is to add a flag to indicate whether the lock is acquired with conflict

Do you mean adding the flag somewhere else?

Reply:
s/WTIH/WITH

Reply:
> > I think more simple approach to do that is to add a flag to indicate whether the lock is acquired with conflict
> 
> Do you mean adding the flag somewhere else?

Yes. Add an extra flag instead of a new lock type. It would be much easier to handle compatibility stuff, I think.

Reply:
@MyonKeminta 
It's updated, PTAL.